file/join.rs

```diff
@@ -0,0 +1,224 @@
+//
+// Copyright (c) 2024 Hemi Labs, Inc.
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+extern crate clap;
+extern crate gettextrs;
+extern crate walkdir;
+
+use clap::Parser;
+use gettextrs::{bind_textdomain_codeset, setlocale, textdomain, LocaleCategory};
+use plib::PROJECT_NAME;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+/// join - relational database operator
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// Additional lines to include when there are no matches
+    #[arg(short, long)]
+    a: Option<u8>,
+
+    /// Replace empty output fields with the specified string
+    #[arg(short, long)]
+    e: Option<String>,
+
+    /// Output fields in specified order
+    #[arg(short, long, value_delimiter = ',')]
+    o: Option<Vec<String>>,
+
+    /// Field separator character
+    #[arg(short, long, default_value_t = ' ')]
+    t: char,
+
+    /// Output only unpairable lines from file_number
+    #[arg(short, long)]
+    v: Option<u8>,
+
+    /// Join on the specified field of file 1
+    #[arg(short = '1', long, default_value_t = 1)]
+    field1: usize,
+
+    /// Join on the specified field of file 2
+    #[arg(short = '2', long, default_value_t = 1)]
+    field2: usize,
+
+    /// File 1
+    file1: PathBuf,
+
+    /// File 2
+    file2: PathBuf,
+}
+
+fn parse_fields(line: &str, sep: char) -> Vec<String> {
+    line.split(sep)
+        .map(|s| s.to_string())
+        .collect()
+}
+
+fn read_file_lines(file_path: &PathBuf, sep: char) -> Result<Vec<Vec<String>>, Box<dyn std::error::Error>> {
+    let file = File::open(file_path).map_err(|_| format!("Unable to open file"))?;
+    let reader = BufReader::new(file);
+    let mut lines = Vec::new();
+
+    for line in reader.lines() {
```

Comment:
Reading the whole file will result in out of memory and crush the app. Reading from files needs to be heavily refactored